src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -707,6 +708,7 @@ public static class Options {
         private User defaultUser;
         private boolean offline = false;
         private final ConfigCatHooks configCatHooks = new ConfigCatHooks();
+        private Function<FilterFunctionParameters, Boolean> logFilterFunction;
```

Comment:
Generally, it's a good idea to encapsulate context-like data into a class instead of passing it as multiple parameters. However, in this case I'd rather not do it because of performance considerations. AFAIK this would lead to heap allocation each time the filter callback is called, which would be nice to avoid.

So I suggest a callback with multiple parameters instead: https://stackoverflow.com/a/27872395

BTW, what is the oldest Java version we support? `Function` seems to be a thing only since Java 8.